packages/nimble-components/src/table-column/tests/table-column.stories.ts

```diff
@@ -0,0 +1,416 @@
+    title: 'Table Columns',
+    decorators: [withXD],
+    parameters: {
+        docs: {
+            description: {
+                component: overviewText
+            }
+        },
+        design: {
+            artboardUrl:
+                'https://xd.adobe.com/view/5b476816-dad1-4671-b20a-efe796631c72-0e14/screen/d389dc1e-da4f-4a63-957b-f8b3cc9591b4/specs/'
+        }
+    },
+    // prettier-ignore
+    render: createUserSelectedThemeStory(html<CommonTableArgs>`
+    ${usageWarning('table')}
+    <${tableTag}
+        ${ref('tableRef')}
+        id-field-name="${idFieldName}"
+        data-unused="${x => x.updateData(x)}"
+    >
+        <${tableColumnTextTag}
+            column-id="first-name-column"
+            field-name="firstName"
+        >
+            First Name
+        </${tableColumnTextTag}>
+        <${tableColumnTextTag}
+            column-id="last-name-column"
+            field-name="lastName"
+        >
+            Last Name
+        </${tableColumnTextTag}>
+        <${tableColumnTextTag}
+            column-id="favorite-color-column"
+            field-name="favoriteColor"
+        >
+            Favorite Color
+        </${tableColumnTextTag}>
+        <${tableColumnTextTag}
+            column-id="quote-column"
+            field-name="quote"
+        >
+            Quote
+        </${tableColumnTextTag}>
+    </${tableTag}>
+    `),
+    argTypes: {
+        tableRef: {
+            table: {
+                disable: true
+            }
+        },
+        updateData: {
+            table: {
+                disable: true
+            }
+        }
+    },
+    args: {
+        tableRef: undefined,
+        updateData: x => {
+            void (async () => {
+                // Safari workaround: the table element instance is made at this point
+                // but doesn't seem to be upgraded to a custom element yet
+                await customElements.whenDefined('nimble-table');
+                x.tableRef.setData(simpleData);
+            })();
+        }
+    }
+};
+
+export default metadata;
+
+// In the Docs tab, Storybook doesn't render the title of the first story
+// This is a placeholder to get the useful ones to render
+export const columns: StoryObj<CommonTableArgs> = {};
+
+type ColumnOrderOption = 'FirstName, LastName' | 'LastName, FirstName';
+
+interface ColumnOrderTableArgs extends CommonTableArgs {
+    columnOrder: ColumnOrderOption;
+}
+
+const columnOrderDescription = `Configure columns by adding column elements as children of the table. 
+The order of the elements controls the order that columns will appear in the table.`;
+
+export const columnOrder: StoryObj<ColumnOrderTableArgs> = {
+    parameters: {
+        docs: {
+            description: {
+                story: columnOrderDescription
+            }
+        }
+    },
+    // prettier-ignore
+    render: createUserSelectedThemeStory(html<ColumnOrderTableArgs>`
+        ${usageWarning('table')}
```

Comment:
I think we should avoid having the usage warning render in the example code text. I'd argue the banner is enough in the rendered component and this just muddles trying to understand how it works. Maybe we can add the code-hide class to it?
![image](https://user-images.githubusercontent.com/1588923/224375543-0bc92895-1ed2-4549-8942-a62b41d975de.png)